app/start_handlers.py

```diff
@@ -123,9 +123,13 @@ async def post(self, request):
                              client_ip=request['client_ip'], client_id=request['client_id'], trace=request['trace'])
                 raise ex
 
-        await remember(uac_json['caseId'], request)
-
-        self.validate_case(uac_json)
+        if uac_json['receiptReceived']:
+            raise InactiveCaseError
+        elif not uac_json['receiptReceived'] and not uac_json['active']:
+            raise ExerciseClosedError
+        else:
+            await remember(uac_json['caseId'], request)
+            self.validate_case(uac_json)
```

Comment:
Not sure about this - it feels like an 
if receiptReceived, 
else 
    if active 
    else
might feel more logical, but not a show stopper

Reply:
You are right. I was debating whether to change this or not before pushing my changes. I've updated it now.